app/lib/actions/katello/product/create.rb

```diff
@@ -34,6 +34,7 @@ def finalize
           product = ::Katello::Product.find(input[:product][:id])
           product.cp_id = input[:cp_id]
           product.save!
+          product.import_subscription(input[:subscription_id])
```

Comment:
ahhh, then why not move ReindexSubscriptions to use finalize instead of run.  That way it'll get executed after it gets the id assigned.

Alternatively we could move the assigning of the ID to the run phase.